### PR TITLE
Dafny VSCode Integration

### PR DESCRIPTION
**Pull request by markusschaden**
Hi everyone

I'm one of the developers of the new Dafny VSCode Plugin, which is already published on the VSCode marketplace. [https://marketplace.visualstudio.com/items?itemName=FunctionalCorrectness.dafny-vscode](Dafny-VSCode). Right know it downloads the Dafny environment from our forked repository, because we needed some extensions. These changes were needed to get an better experience for users. 

Most important are the verbs symbols and counterModel. The first one returns a symboltable of a program. That contains local variables, methods, method references and everything needed to implement IntelliSense, Go To Definition, Rename and CodeLens. 
The later one sends back the countermodel if a proof does not hold. Both of them are returning their answers json encoded, for easier parsing.

The last feature which we implemented in the plugin is a version check. To allow that the server has to return his current version number, to calculate if there is a newer version available on GitHub. Right know the class VersionCheck is used. (Guess, based on the naming, couple of weeks ago this class also did the check). This can be definitely optimized so that it picks the version from assemblyversion. It just must be SemVer compatible. 
The check itself is done in our plugin and could be adjusted, to point to your repository, as soon as our pull request is accepted and a release is created. For that the version needs to be greater than 1.9.16, otherwise no update notification would be shown. 

If you have any questions just let me know. 

Best regards
Markus

### CODE REVIEW

**Comment by cpitclaudel** [comment]
Hi there. I wrote the Dafny server. This looks great! But, can you please document these new commands? Ideally as a PR on the existing docs :)

**Comment by cpitclaudel** [comment]
@markusschaden Why does this PR disable caching? (and, did you test that the Emacs mode still works with your changes?).

File: Source/DafnyServer/Utilities.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ class ServerUtils {
       Dafny.DafnyOptions.O.ProverKillTime = 10; //This is just a default; it can be overriden
 
       if (CommandLineOptions.Clo.Parse(args)) {
-        DafnyOptions.O.VerifySnapshots = 2; // Use caching
+        DafnyOptions.O.VerifySnapshots = 0;
```
Comment (cpitclaudel) [inline Source/DafnyServer/Utilities.cs]: This looks wrong… ? (Thanks @parno for spotting this)

**Comment by markusschaden** [comment]
@cpitclaudel sorry for that. We had some troubles during testing with caching enabled. After some time the verification result wasn't correct. The lines where errors where found were incorrect

**Comment by cpitclaudel** [comment]
Yes, this is a known issue.  I think @RustanLeino was going to work on it at some point.
If you don't use caching, there's little use for a verification server anyway: you can just send the entire file to Dafny on the command line, and parse the output.

**Pull request merged**